pkg/component/plutono/plutono.go

```diff
@@ -84,6 +161,546 @@ func (p *plutono) WaitCleanup(ctx context.Context) error {
+			return nil, err
+		}
+
+		// Need stable order before passing the dashboards to Plutono config to avoid unnecessary changes
+		kubernetesutils.ByName().Sort(existingConfigMaps)
+
+		// Read extension monitoring configurations
+		for _, cm := range existingConfigMaps.Items {
+			if operatorsDashboard, ok := cm.Data[v1beta1constants.PlutonoConfigMapOperatorDashboard]; ok && operatorsDashboard != "" {
+				extensionsDashboards.WriteString(fmt.Sprintln(strings.ReplaceAll(strings.ReplaceAll(operatorsDashboard, "Grafana", "Plutono"), "loki", "vali")))
+			}
+			if usersDashboard, ok := cm.Data[v1beta1constants.PlutonoConfigMapUserDashboard]; ok && usersDashboard != "" {
+				extensionsDashboards.WriteString(fmt.Sprintln(strings.ReplaceAll(strings.ReplaceAll(usersDashboard, "Grafana", "Plutono"), "loki", "vali")))
+			}
+		}
+
+		if extensionsDashboards.Len() > 0 {
+			if err := yaml.Unmarshal([]byte(extensionsDashboards.String()), &dashboards); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	for dashboardPath, dashboardEmbed := range requiredDashboards {
+		if err := fs.WalkDir(dashboardEmbed, dashboardPath, func(path string, dirEntry fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			normalizedPath := strings.TrimPrefix(strings.TrimPrefix(path, dashboardPath), "/")
+			if normalizedPath == "" {
+				// No need to process top level.
+				return nil
+			}
+
+			// Normalize to / since it will also work on Windows
+			normalizedPath = filepath.ToSlash(normalizedPath)
+
+			if dirEntry.IsDir() {
+				if len(sets.New[string](strings.Split(path, "/")...).Intersection(ignorePaths)) > 0 {
+					return filepath.SkipDir
+				}
+				return nil
+			}
+
+			data, err := dashboardEmbed.ReadFile(path)
+			if err != nil {
+				return fmt.Errorf("error reading %s: %s", normalizedPath, err)
+			}
+			dashboards[normalizedPath[strings.LastIndex(normalizedPath, "/")+1:]] = string(data)
+
+			return nil
+		}); err != nil {
+			return nil, err
+		}
+	}
+
+	// this is necessary to prevent hitting configmap size limit.
+	return convertToCompactJSON(dashboards)
+}
+
+func (p *plutono) getService() *corev1.Service {
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: p.namespace,
+			Labels:    getLabels(),
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name:     "web",
+					Port:     port,
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+			Selector: getLabels(),
+		},
+	}
+
+	if p.values.ClusterType == component.ClusterTypeSeed {
+		service.Labels = utils.MergeStringMaps(service.Labels, map[string]string{v1beta1constants.LabelRole: v1beta1constants.LabelMonitoring})
+	}
+
+	return service
+}
+
+func (p *plutono) getDeployment(providerConfigMapName, dataSourceConfigMapName, dashBoardConfigMapName string) *appsv1.Deployment {
+	sizeLimit := resource.MustParse("100Mi")
```

Comment:
Inline this with the help of https://github.com/gardener/gardener/blob/55123e8724bca6939bb5a367d87ac432c0568c61/pkg/utils/miscellaneous.go#L135-L138